utils/gen.js

```diff
@@ -7,9 +7,14 @@
 const { writeFileSync } = require('fs');
 const { join } = require('path');
 
-const { versions, platform, arch } = process;
+const { versions, platform, arch, env } = process;
 const file = join(__dirname, '..', 'turbo-cache-key.json');
 const node = versions.node.split('.')[0];
-const str = JSON.stringify({ node, platform, arch });
+const str = JSON.stringify({
+  node,
+  platform,
+  arch,
+  imageVersion: env.ImageVersion ?? 'local',
```

Comment:
**suggestion (non-blocking):** ‏It might nice to comment what the `ImageVersion` looks like. For example:

```suggestion
  // the GitHub runner image version (example: "20231126.1.1")
  imageVersion: env.ImageVersion ?? 'local',
```